src/httpserver.cpp

```diff
@@ -296,9 +296,12 @@ static bool HTTPBindAddresses(struct evhttp* http)
     std::vector<std::pair<std::string, uint16_t> > endpoints;
 
     // Determine what addresses to bind to
-    if (!gArgs.IsArgSet("-rpcallowip")) { // Default to loopback if not allowing external IPs
+    if (!(gArgs.IsArgSet("-rpcallowip") && gArgs.IsArgSet("-rpcbind"))) { // Default to loopback if not allowing external IPs
```

Comment:
In commit `net: Always default rpcbind to localhost, never "all interfaces"`:

Nit: I think `if !a || !b` is easier on the eyes than `if !(a && b)`.